chromeapp/app/scripts/dnssd/dns-sd.js

```diff
@@ -0,0 +1,722 @@
+/*jshint esnext:true*/
+/* globals Promise */
+'use strict';
+
+/**
+ * The client API for interacting with mDNS and DNS-SD.
+ *
+ * This is based in part on the Bonjour APIs outlined in 'Zero Configuration
+ * Networking: The Definitive Guide' by Cheshire and Steinberg in order to
+ * provide a familiar interface.
+ *
+ * 'RFC 6762: Multicast DNS' is the model for many of the decisions and actions
+ * take in this module. 'The RFC' in comments below refers to this RFC. It can
+ * be accessed here:
+ *
+ * https://tools.ietf.org/html/rfc6762#
+ *
+ * Since this is programming to a specification (or at least to an RFC), it is
+ * conforming to a standard. Actions are explained in comments, with direct
+ * references to RFC sections as much as is possible.
+ */
+
+
+var dnsUtil = require('./dns-util');
+var dnsController = require('./dns-controller');
+var dnsCodes = require('./dns-codes');
+var resRec = require('./resource-record');
+var dnsPacket = require('./dns-packet');
+
+var MAX_PROBE_WAIT = 250;
```

Comment:
Suggest adding comments for these constants - what they mean, where they came from, are they required by spec, etc.